xml/System.Runtime.CompilerServices/Unsafe.xml

```diff
@@ -370,8 +441,18 @@
         <typeparam name="T">The type which the object will be cast to.</typeparam>
         <param name="o">The object to cast.</param>
         <summary>Casts the given object to the specified type.</summary>
-        <returns>The original object, casted to the given type.</returns>
-        <remarks>To be added.</remarks>
+        <returns>The original object, cast to the given type.</returns>
+        <remarks>
+          <format type="text/markdown"><![CDATA[
+
+## Remarks
+
+This API is used to cast an object to the given type, suppressing the runtime's normal type safety checks. It is the caller's responsibility to ensure that the cast is legal. No `InvalidCastException` will be thrown.
+
+The behavior of `Unsafe.As<T>(o)` is only well-defined if the typical "safe" casting operation `(T)o` would have succeeded. Use of this API to circumvent casts that would otherwise have failed is unsupported and could result in runtime instability.
```

Comment:
```suggestion
The behavior of `Unsafe.As<T>(o)` is only well-defined if the typical "safe" casting operation `(T)o` would have succeeded. Use of this API to circumvent casts that would otherwise have failed is unsupported and could result in runtime instability.
One could use `Debug.Assert(o is T)` to have an assertion at least in Debug-builds.
```
Or similar.

I see that quite a lot of people aren't aware of the possibilities of `Debug.Assert`, so maybe it's a good idea to give a pointer (😉) to that use in this context.